homeassistant/components/modbus/__init__.py

```diff
@@ -121,6 +127,29 @@ def number(value: Any) -> int | float:
         raise vol.Invalid(f"invalid number {value}") from err
 
 
+def control_scan_interval(config: dict) -> dict:
+    """Control scan_interval."""
+    for hub in config:
+        minimum_scan_interval = DEFAULT_SCAN_INTERVAL
+        for component, conf_key in CONF_PLATFORMS:
+            if conf_key in hub:
+                for entry in hub[conf_key]:
+                    scan_interval = entry.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+                    if scan_interval < MINIMUM_SCAN_INTERVAL:
+                        _LOGGER.warning(
+                            "%s %s scan_interval(%d) is adjusted to minimum(%d)",
+                            component,
+                            entry.get(CONF_NAME),
+                            scan_interval,
+                            MINIMUM_SCAN_INTERVAL,
+                        )
+                        scan_interval = MINIMUM_SCAN_INTERVAL
+                    entry[CONF_SCAN_INTERVAL] = scan_interval
+                    scan_interval = min(scan_interval, minimum_scan_interval)
```

Comment:
This line doesn't seem needed. Should it be:
```suggestion
                    minimum_scan_interval = min(scan_interval, minimum_scan_interval)
```

Reply:
ups ! DONE.